API_GUIDE.md

```diff
@@ -138,9 +138,8 @@ There are three differences between this multidevice snippet and the previous
 single device snippet:
 
 - `xmp.spawn()` creates the processes that each run an XLA device.
-- `ParallelLoader` loads the training data onto each device.
-- `xm.optimizer_step(optimizer)` no longer needs a barrier. ParallelLoader
-automatically creates an XLA barrier that evaluates the graph.
+- `MpDeviceLoader` loads the training data onto each device.
+- `xm.optimizer_step(optimizer)` creates an XLA barrier that evaluates the graph and update the model's parameters.
```

Comment:
The description for `xm.optimizer_step()` should stay as is as it's still the `pl.MpDeviceLoader` that ends up marking the step (https://github.com/pytorch/xla/blob/master/torch_xla/distributed/parallel_loader.py#L37), unless `xm.optimizer_step` is called with `barrier` kwarg.

Reply:
@jysohn23 Since the previous [example](https://github.com/lezwon/xla/blob/code_desc_update/API_GUIDE.md#running-on-a-single-xla-device)  for running on a single xla device no longer uses `xm.optimizer_step(optimizer, barrier=True)` it wouldn't make sense to have  _`xm.optimizer_step(optimizer)` no longer needs a barrier._ in the doc. Would something like _`xm.optimizer_step(optimizer)` consolidates the gradients between cores and issues the XLA device step computation_ work?